falcon/util/uri.py

```diff
@@ -192,7 +192,7 @@ def decode(encoded_uri):
 
         # PERF(kgriffs): Don't take the time to instantiate a new
         # string unless we have to.
-        if '+' in decoded_uri:
+        if unquote_plus and '+' in decoded_uri:
```

Comment:
Since there is a `PERF` note above, we might want to try to optimize this as much as possible.

As `unquote_plus=False` path is used mostly in testing, we might want to put `'+' in decoded_uri` first to short-circuit `AND`.